src/components/editor/Bindings/SchemaInference/Button.tsx

```diff
@@ -1,24 +1,105 @@
 import { DataObject } from '@mui/icons-material';
 import { Button, Skeleton } from '@mui/material';
 import SchemaInferenceDialog from 'components/editor/Bindings/SchemaInference/Dialog';
-import { useBindingsEditorStore_collectionData } from 'components/editor/Bindings/Store/hooks';
+import {
+    useBindingsEditorStore_collectionData,
+    useBindingsEditorStore_setDocumentsRead,
+    useBindingsEditorStore_setInferredSpec,
+    useBindingsEditorStore_setLoadingInferredSchema,
+} from 'components/editor/Bindings/Store/hooks';
 import { useEntityWorkflow } from 'context/Workflow';
+import { isEmpty } from 'lodash';
 import { useState } from 'react';
 import { FormattedMessage } from 'react-intl';
+import { useLocalStorage, useUnmountPromise } from 'react-use';
+import getInferredSchema from 'services/schema-inference';
+import { useResourceConfig_currentCollection } from 'stores/ResourceConfig/hooks';
+import {
+    getStoredGatewayAuthConfig,
+    LocalStorageKeys,
+} from 'utils/localStorage-utils';
 
 function SchemaInferenceButton() {
     const workflow = useEntityWorkflow();
 
     // Bindings Editor Store
     const collectionData = useBindingsEditorStore_collectionData();
 
+    const setInferredSpec = useBindingsEditorStore_setInferredSpec();
+
+    const setDocumentsRead = useBindingsEditorStore_setDocumentsRead();
+
+    const setLoadingInferredSchema =
+        useBindingsEditorStore_setLoadingInferredSchema();
+
+    // Resource Config Store
+    const currentCollection = useResourceConfig_currentCollection();
+
     const [open, setOpen] = useState<boolean>(false);
 
+    const [gatewayConfig] = useLocalStorage(
+        LocalStorageKeys.GATEWAY,
+        getStoredGatewayAuthConfig()
+    );
+
+    const resolveWhileMounted = useUnmountPromise();
+
     const openSchemaInferenceDialog = (
         event: React.MouseEvent<HTMLElement>
     ) => {
         event.preventDefault();
 
+        setLoadingInferredSchema(true);
+
+        if (currentCollection && gatewayConfig?.gateway_url && collectionData) {
+            resolveWhileMounted(
+                getInferredSchema(gatewayConfig, currentCollection)
+            )
+                .then(
```

Comment:
This gives me a _tiny pause_ since this is the response of a hook being called conditionally and wondered if we were using a foot gun. However, I think it is fine since it is returning a memoized promise.